files/scripts/syncd.sh

```diff
@@ -75,6 +98,17 @@ function waitplatform() {
 
 function stopplatform1() {
 
+    if [[ x"$(${SONIC_DB_CLI} STATE_DB GET ORCH_ABRT_STATUS)" == x"1" ]]; then
+        # Collecting saisdkdump before restarting syncd
+        # Runs when orchagent is aborted because of SAI failure.
+        # Only enabled for mellanox platform
+        if [[ x$sonic_asic_platform == x"mellanox" ]]; then
+            collect_saisdkdump
```

Comment:
Can you run this with timeout? In case if saisdkdump is stuck it means syncd restart would also be stuck

Reply:
Good catch, will update

Reply:
Updated